.github/workflows/requirements/build-requirements.in

```diff
@@ -0,0 +1,4 @@
+pip
```

Comment:
I think this file needs a comment + a comment in pyproject.toml, remdining people that they need to be kept in sync appropriately.

Reply:
Great call! I'll do that, also update the build-requirements.in to have the same versions as `pyproject.toml`.

However, I'm not familiar with the `platform_python_implementation != 'PyPy'` command, written [here](https://github.com/pyca/cryptography/blob/main/pyproject.toml#L7) on the `pyproject.toml`. Should I also include it on the build-requirements.in ?

Reply:
While I did the other changes, I tested including this `platform_python_implementation != 'PyPy'` to build-requirements.in and it worked fine! So I took the liberty and added it, now it's all synced with pyproject.toml =)